### ISSUE TITLE
dist build / tsc failing with empty project

### ISSUE THREAD

**Issue by timojokinen** [vendure-ecommerce/vendure#286]
**Describe the bug**
tsc fails with an empty project.
There are errors in multiple files all concerning translation types

Example:
```
node_modules/@vendure/core/dist/entity/collection/collection.entity.d.ts:25:5 - error TS2416: Property 'translations' in type 'Collection' is not assignable to the same property in base type 'Translatable'.
  Type 'Translation<Collection>[]' is not assignable to type 'Translation<any>[]'.
    Type 'Translation<Collection>' is not assignable to type 'Translation<any>'.
      Type 'Translation<Collection>' is not assignable to type '{ [x: string]: string; }'.
        Property 'id' is incompatible with index signature.
          Type 'string | number' is not assignable to type 'string'.
            Type 'number' is not assignable to type 'string'.

     translations: Array<Translation<Collection>>;
```

**To Reproduce**
1. Install vendure project with @vendure/create
2. Settings: SQLite / TypeScript / Populate with sample data
3. npm i / yarn
4. npm run build / yarn build

**Environment (please complete the following information):**
 - @vendure/core version: 0.10.1
 - Nodejs version: v12.13.0
 - Database (mysql/postgres etc): SQLite
 - tsc: 6.14.4

**Comment by michaelbromley** [comment]
This is likely due to an incompatibility with TypeScript 3.8. Try to specify your TS version as 3.7.5 and try again.

In the next minor release I'll update to TS 3.8.

**Comment by timojokinen** [comment]
I installed typescript@3.7.5 locally and the build worked thanks.
Maybe the cli should install the correct version locally by default

**Comment by michaelbromley** [comment]
Yes, I think it is a good idea since TypeScript doesn't really follow semantic versioning.

**Comment by michaelbromley** [comment]
Fixed in a2fba131a6e10763eaf2d032fbd556b6bad08af1
